Extensions/NotificationService/NotificationService.swift

```diff
@@ -11,42 +11,60 @@ private let log = Logger.browserLogger
 private let CategorySentTab = "org.mozilla.ios.SentTab.placeholder"
 
 class NotificationService: UNNotificationServiceExtension {
-    var display: SyncDataDisplay!
-    lazy var profile: ExtensionProfile = {
-        let profile = ExtensionProfile(localName: "profile")
-        return profile
-    }()
+    var display: SyncDataDisplay?
+    var profile: ExtensionProfile?
 
     // This is run when an APNS notification with `mutable-content` is received.
     // If the app is backgrounded, then the alert notification is displayed.
     // If the app is foregrounded, then the notification.userInfo is passed straight to
     // AppDelegate.application(_:didReceiveRemoteNotification:completionHandler:)
     // Once the notification is tapped, then the same userInfo is passed to the same method in the AppDelegate.
     override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        let userInfo = request.content.userInfo
-        if Logger.logPII && log.isEnabledFor(level: .info) {
-            // This will be visible in the Console.app when a push notification is received.
-            NSLog("NotificationService APNS NOTIFICATION \(userInfo)")
-        }
+        DispatchQueue.global().async {
```

Comment:
What is the reason for wrapping it in an async operation? Is that really going to make a difference? We still need to finish this amount of work in the same (undefined) period of time and then run the callback.

Reply:
This is mainly just uncertainty on my part as to when the kill timer starts. This is to ensure we get through this on the main thread as quickly as possible.